### DISCUSSION TITLE
What is the standalone usecase for metricflow / Can metricflow be used without dbt cloud/semantic layer?

### DISCUSSION THREAD

**Discussion by janmechtel** [dbt-labs/metricflow#734] [Q&A] [answered]
I'm trying to understand what the output of metricflow is.

On the hand I found many times that ones needs to have a dbt team/enterprise license and use dbt cloud to be able to query the metrics.

On the other hand Metricflow is described as a way to define metrics. How can one consume these metrics? For example is there a way to materialize certain metrics into a table? 

This way having the definition of the metric in Metricflow/code would be much more readable and structured and would avoid to repeat the same definitions for different granularities/dimensions.

**Comment by ghost** [comment]
+1 would love clarification on the forecasted integration with dbt-core

**Comment by mirnawong1** [comment, marked as answer]
hey @janmechtel , that's a great question! I'll do my best to answer them cc @Jstein77 

The dbt Semantic Layer is proprietary, however, some components of the dbt Semantic Layer are open source, like dbt-core and MetricFlow.

dbt Cloud Developer plans or dbt Core users can define metrics in their project, including a local dbt Core project, using the dbt Cloud IDE or the command line. You can do that in dbt Core and query your data warehouse locally using the [MetricFlow commands](https://docs.getdbt.com/docs/build/metricflow-commands) (e..g `mf`) from your local dbt project. 

However, to consume (or query metrics dynamically) metrics - you need the dbt Cloud API access and integration, which are available on a dbt Cloud Team or Enterprise plan.

We have an [architecture page](https://docs.getdbt.com/docs/use-dbt-semantic-layer/sl-architecture#dbt-semantic-layer-components) that provides a table description.

**Reply by janmechtel** [reply]
Thank you, I'm not sure how i missed those parts of the documentation. I searched quite thoroughly.

I'll look into whether we could use those metrics and maybe use dbt-metabase to sync them to metabase.
